server/src/main/java/io/druid/metadata/SQLMetadataStorageActionHandler.java

```diff
@@ -206,306 +200,268 @@ public Boolean withHandle(Handle handle) throws Exception
+                              r.getBytes("status_payload"),
+                              statusType
+                          );
+                        }
+                        catch (IOException e) {
+                          log.makeAlert(e, "Failed to parse status payload")
+                             .addData("entry", r.getString("id"))
+                             .emit();
+                          throw new SQLException(e);
+                        }
                       }
                     }
-                  }
-              ).list();
+                ).list();
+          }
         }
-      }
     );
   }
 
   public boolean addLock(final String entryId, final LockType lock)
   {
-    return retryingHandle(
-      new HandleCallback<Boolean>()
-      {
-        @Override
-        public Boolean withHandle(Handle handle) throws Exception
+    return connector.retryWithHandle(
+        new HandleCallback<Boolean>()
         {
-          return handle.createStatement(
-                      String.format(
-                          "INSERT INTO %1$s (%2$s_id, lock_payload) VALUES (:entryId, :payload)",
-                          lockTable, entryTypeName
-                      )
-                  )
-                       .bind("entryId", entryId)
-                       .bind("payload", jsonMapper.writeValueAsBytes(lock))
-                       .execute() == 1;
+          @Override
+          public Boolean withHandle(Handle handle) throws Exception
+          {
+            return handle.createStatement(
+                String.format(
+                    "INSERT INTO %1$s (%2$s_id, lock_payload) VALUES (:entryId, :payload)",
+                    lockTable, entryTypeName
+                )
+            )
+                         .bind("entryId", entryId)
+                         .bind("payload", jsonMapper.writeValueAsBytes(lock))
+                         .execute() == 1;
+          }
         }
-      }
     );
   }
 
   public boolean removeLock(final long lockId)
   {
-    return retryingHandle(
-      new HandleCallback<Boolean>()
-      {
-        @Override
-        public Boolean withHandle(Handle handle) throws Exception
+    return connector.retryWithHandle(
+        new HandleCallback<Boolean>()
         {
-          return handle.createStatement(
-                      String.format(
-                          "DELETE FROM %s WHERE id = :id",
-                          lockTable
-                      )
-                  )
-                       .bind("id", lockId)
-                       .execute() == 1;
+          @Override
+          public Boolean withHandle(Handle handle) throws Exception
+          {
+            return handle.createStatement(
+                String.format(
+                    "DELETE FROM %s WHERE id = :id",
+                    lockTable
+                )
+            )
+                         .bind("id", lockId)
+                         .execute() == 1;
```

Comment:
This doesn't seem right. If it succeeded in the delete but simply failed to register the confirmation, then this will always be failed even if it was deleted the first time.